### PR TITLE
Branch v1.2b set one zy

### LINKED ISSUE #60: As a second-time user, I can set one specific ingredient level to a specified amount
... so that I can adjust ingredients levels according to my store's current situation to better use tCheck to track the ingredients levels.

### PR DESCRIPTION
**Pull request by Diwu-Yi**
Complete backend code for Set Command, however, Ui is not working.
Abstract out ingredient book.
Next step : add tests. #

### CODE REVIEW

File: src/main/java/seedu/address/model/IngredientBook.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+     * Returns true if a person with the same identity as {@code person} exists in the address book.
+     */
+    public boolean hasIngredient(Ingredient ingredient) {
+        requireNonNull(ingredient);
+        return ingredients.contains(ingredient);
+    }
+
+
+    public void setIngredient(Ingredient target, Ingredient newAmount) {
+        requireNonNull(newAmount);
```
Comment (ureshiiYing) [inline src/main/java/seedu/address/model/IngredientBook.java:67]: Should `target` be non null too?

Comment (Diwu-Yi) [reply src/main/java/seedu/address/model/IngredientBook.java:67]: ya, I will check it as well !

**Review by WM71811 — DISMISSED** [review]
LGTM

**Review by Persdre — DISMISSED** [review]
LGTM

**Review by ureshiiYing — APPROVED** [review]
LGTM

**Review by WM71811 — APPROVED** [review]
LGTM

**Pull request merged**